### PR TITLE
Most-59: e2e tests

### PR DESCRIPTION
**Pull request by lesniak43**
e2e-tests for cross-chain transactions.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -306,8 +306,17 @@ e2e-test:
 .PHONY: e2e-tests
 e2e-tests: # Run e2e tests
 e2e-tests:
+	# TEST_CASE=eth_to_azero::eth_to_weth make e2e-test # not working
```
Comment (obrok) [inline Makefile:309]: Please add comment either here or in some other prominent place, explaining that the tests are order-dependent

Comment (lesniak43) [reply Makefile]: done

File: e2e-tests/src/balance.rs [hunk 2]
```diff
@@ -0,0 +1,216 @@
+use anyhow::{anyhow, Result};
+use ethers::types::U256;
+
+/// Struct holding the current balances on both chains for a given account pair.
+/// It includes the native coins, and all the wrapped tokens we're concerned with.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct Balance {
```
Comment (obrok) [inline e2e-tests/src/balance.rs:7]: :+1: nice. I'd add a comment that the impl serves as a model for the transfers we care about.

File: e2e-tests/src/test/azero_to_eth.rs [hunk 3]
```diff
@@ -1,93 +1,197 @@
...
+}
+
+/// One-way `Aleph Zero` -> `Ethereum` transfer through `most`.
+/// Requires a prior transaction in the other direction to have completed.
+/// This is easily done by running the test for the other direction first.
+/// Approves the `most` contract to use the wETH funds.
+/// Burns the required funds in the wETH contract on Aleph Zero.
+/// Transfers `transfer_amount` of burned wETH over the bridge, the Ethereum account receives native ETH.
+/// Waits for the transfer to complete.
+/// It relies on all the relevant contracts being deployed on both ends and the (wETH_ETH:wETH_AZERO) pair having been added to `most`.
```
Comment (obrok) [inline e2e-tests/src/test/azero_to_eth.rs:136]: Suggestion: Separate actions from assertions in these comments. These descriptions make no differentiation between what the test does and what it expects to happen as a result. "Approves" is written in the same way as "Burns", but only one of these actions is taken by the test driver, the other is what MOST is supposed to do as a result. Or just remove these altogether, I think they carry about the same information as the body of the test anyway.

**Review by obrok — APPROVED** [review]

File: e2e-tests/src/test/azero_to_eth.rs [hunk 4]
```diff
@@ -1,93 +1,197 @@
...
 /// One-way `Aleph Zero` -> `Ethereum` transfer through `most`.
 /// Requires a prior transaction in the other direction to have completed.
 /// This is easily done by running the test for the other direction first.
 /// Approves the `most` contract to use the wETH funds.
 /// Burns the required funds in the wETH contract on Aleph Zero.
-/// Transfers `transfer_amount` of burned wETH over the bridge, unwrapping the transfer to a specified Ethereum account.
+/// Transfers `transfer_amount` of burned wETH over the bridge.
 /// Waits for the transfer to complete.
-/// Verifies that the correct amount of ETH is present on the Ethereum chain.
 /// It relies on all the relevant contracts being deployed on both ends and the (wETH_ETH:wETH_AZERO) pair having been added to `most`.
```
Comment (kostekIV) [inline e2e-tests/src/test/azero_to_eth.rs:14]: Suggestion: use numbered list here for test steps.

File: Makefile [hunk 5]
```diff
@@ -304,10 +304,18 @@ e2e-test:
 		cargo test test::$(TEST_CASE) -- --color always --exact --nocapture --test-threads=1
 
 .PHONY: e2e-tests
-e2e-tests: # Run e2e tests
+e2e-tests: # Run cross-chain transfer e2e tests. All tests must be run without interruption, the order is important.
```
Comment (kostekIV) [inline Makefile:307]: Lest add an info here what the e2e-tests requires to be run. Also add section to the project readme how to run e2e tests locally.

**Review by kostekIV — APPROVED** [review]

File: e2e-tests/src/balance.rs (hunk 2 above)
Comment (lesniak43) [reply e2e-tests/src/balance.rs:9]: done

File: e2e-tests/src/test/azero_to_eth.rs (hunk 3 above)
Comment (lesniak43) [reply e2e-tests/src/test/azero_to_eth.rs]: removed the steps from docs altogether

File: e2e-tests/src/test/azero_to_eth.rs (hunk 4 above)
Comment (lesniak43) [reply e2e-tests/src/test/azero_to_eth.rs]: removed the steps altogether, as @obrok suggested

File: Makefile (hunk 5 above)
Comment (lesniak43) [reply Makefile]: done

File: e2e-tests/src/test/azero_to_eth.rs (hunk 3 above)
Comment (lesniak43) [reply e2e-tests/src/test/azero_to_eth.rs]: I believe I've managed to find a compromise.

**Comment by lesniak43** [comment]
https://github.com/Cardinal-Cryptography/most/pull/215

**Pull request closed without merging**